Support ignoring actions

pasih opened the pull request:
Simple implementation of ignored actions. Pass `{ ignoreActions: ['someAction'] }` in the plugin options. The ignored actions are simply left out of the undo history.

The part I'm not sure about is EMPTY_STATE. The user of the plugin will have to manually "pass through" the ignored part of the state:

```
  emptyState(state) {
      this.replaceState({ slice1: [], someOtherSlice: state.someOtherSlice });
    },
```

slice1 will be reset and someOtherSlice is assumed to be managed by ignored actions. This is fine to an extent but obviously actions don't map 1:1 to parts of state.

Commit:
Support ignoring actions

anthonygore commented:
Hi!

Good idea. There are two bits of feedback I have:

1) Rename it from ignoreAction to ignoreMutation. "action" has a separate meaning in Vuex.
2) Make it so the options are optional. I think there will be an uncaught error if options is undefined

Would you also consider updating the README to explain usage?

Commit:
Rename ignoreAction to ignoreMutations

Commit:
Document ignoreMutations in README

pasih commented:
I've renamed ignoreAction to ignoreMutations (I figured plural is a good idea because it's expected to be an array). The options object is now initialized to `{}` which should take care of the undefined option scenario.

I've updated the README too (I included it in the API section and added a paragraph about usage). 

It might be worth adding some tests too. I might take a look at that later. Do you have any test runner / assertion library preference?

anthonygore commented:
Great work, thanks a lot!

I don't have any preference for a test library, what do you suggest?
